packages/mobile/src/screens/edit-track-screen/components/SpecialAccessAvailability.tsx

```diff
@@ -115,11 +115,13 @@ export const SpecialAccessAvailability = ({
     ? { follow_user_id: currentUserId }
     : null
   const [selectedSpecialAccessGate, setSelectedSpecialAccessGate] = useState(
-    !('nft_collection' in (initialPremiumConditions ?? {}))
-      ? initialPremiumConditions ?? defaultSpecialAccess
+    !('nft_collection' in (previousPremiumConditions ?? {}))
```

Comment:
lets see if we can use formixContext for this as well?

Reply:
this here is different, it's not always based on the formik values as we keep track of previously selected nft/tip/follow gates as the user switches back and forth.
also, there is some significant logic for disabled states when a user edits a track, based on what the initial availability was. logic is a bit more gnarly than just what's currently on the form